### DISCUSSION TITLE
Workaround for create_tab

### DISCUSSION THREAD

**Discussion by bertradio** [sxyazi/yazi#911] [1. Q&A]
I found what appears to be a bug in tab_create and found an easy workaround.

I like to see the directory name in the tab label, so I set tab_width in theme.toml to 25.

The issue I found is that when I create a new tab, it does not have the directory name in it. If I cd to another direcotory the name of the new dirctory appears. The simple workaround is to define the keymap as :

run = ["tab_create", "cd ."]

**Comment by sxyazi** [comment]
Sounds like a bug, would you like to file a bug report for it?

**Comment by bertradio** [comment]
Sure. Anything I can do to help. I filed a bug report

On another subject, I use yazi on several machines and need to have slightly different keymaps on them. I sync the files now (for convenience), but would like to be able to do something so that yazi senses which machine it is running on and adjusts the keymaps. They are both all running Linux. I looked at toml and it does not seem to be able to access the hostname. Lua can do this, and I use it in init.lua for some changes, depending on which machine is running yazi. Is there a way to set a keymap in lua? If so, then I could do what I need in init.lua. Same for openers in yazi.

Another approach would be to expand on the "for = OS" in yazi. If for could take something like ["linux", 'hostname"] where hostname was optional, that way openers and perhaps keymaps could be customized for particular machines. This would be helpful if directory structures and available programs are different for different machines,

Thanks again for the great work!

**Reply by sxyazi** [reply]
Binding key mappings in Lua is not supported atm, but you can bind a Lua script to the same key for different hosts to perform different tasks. Could you please tell me your common use cases? This would be helpful for me as a design reference, as I hadn't considered the issue of different key mappings on different machines.

**Comment by bertradio** [comment]
I have Yazi on 2 computers: Chromebook (crostini) and Linux Mint desktop. I synchronize my yazi config files between the two.

The directory structure is different on the 2 computers. So if I want to set shortcuts to cd various places that presents a problem. My workaround has been to create links in each computer to various directories. The link names are the same on both computers. So for example, I have a link .Temp on each pointing to the temp directory on each. If I make a shortcut to cd .Temp it works on both.

The other situation is when I have different software on both computers, mostly because of licensing issues. So the open configs on both would need to be different as well. I could get around this again by creating links but it's kind of kludgy.

Neither of these is critical, but it would be nice for yazi to be able to sense the name of the local host and behave appropriately.

Thanks again for a wonderful piece of software!

**Comment by sxyazi** [comment]
Binding key mappings in Lua is now supported in https://github.com/sxyazi/yazi/pull/4031
